process/block/sovereignChainBlock.go

```diff
@@ -2002,7 +2032,79 @@ func (scbp *sovereignChainBlockProcessor) saveSovereignMetricsForCommittedBlock(
 // RestoreBlockIntoPools restores block into pools
 func (scbp *sovereignChainBlockProcessor) RestoreBlockIntoPools(header data.HeaderHandler, body data.BodyHandler) error {
 	scbp.restoreBlockBody(header, body)
-	scbp.blockTracker.RemoveLastNotarizedHeaders()
+
+	// TODO: MX-16507: check how/if to restore incoming scrs/cross chain txs once we have a testnet setup for it
+	// 1. For incoming scrs We should probably have something similar to (sp *shardProcessor) rollBackProcessedMiniBlocksInfo
```

Comment:
actually we shouldn't really have any rollbacks anymore :D 

Like with one shot finality. And we can aim for that as Andromeda is going to be released.

Merging Andromeda into this code, and what is once voted, it is a finished block.

Reply:
That's super awesome, actually, would simplify things a lot !